src/System.Private.CoreLib/shared/System/Runtime/InteropServices/ArrayWithOffset.cs

```diff
@@ -4,19 +4,50 @@
 
 using System.Runtime.CompilerServices;
 
+#if BIT64
+using nuint = System.UInt64;
+#else
+using nuint = System.UInt32;
+#endif
+
 namespace System.Runtime.InteropServices
 {
     public struct ArrayWithOffset
     {
+        private object m_array;
+        private int m_offset;
+        private int m_count;
+
         // From MAX_SIZE_FOR_INTEROP in mlinfo.h
         private const int MaxSizeForInterop = 0x7ffffff0;
 
         public ArrayWithOffset(object array, int offset)
         {
+            int totalSize = 0;
+            if (array != null)
+            {
+                if (!(array is Array arrayObj) || (arrayObj.Rank != 1) || !Marshal.IsBlittable(arrayObj))
+                {
+                    throw new ArgumentException(SR.Argument_NotIsomorphic);
+                }
+
+                nuint nativeTotalSize = (nuint)arrayObj.LongLength * (nuint)arrayObj.GetElementSize();
```

Comment:
For at least the 32-bit case, should this be checked?

Reply:
We have a valid array, so the result has to fix into memory space - it cannot overflow.

Reply:
Ah, of course.